src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+     *
+     * @param command The user's command to be processed.
+     * @throws DukeException If the command is not recognised, or error
+     */
+    public void executeCommand(String command, String filePath) throws DukeException {
+        String[] separateCommand = command.split(" ");
+        System.out.println("--------------------------");
+        System.out.println("--------------------------");
+        if (command.equals("list")) {
+            System.out.println("Task List:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println(" " + (i + 1) + ". " + tasks.get(i).toString());
+            }
+        } else if (command.startsWith("mark") || command.startsWith("unmark")) {
+            try {
+                if (separateCommand.length > 2 || Integer.parseInt(separateCommand[1]) > tasks.size()) {
+                    throw new DukeException("☹ OOPS!!! Invalid number");
+                }
+                int taskNumber = Integer.parseInt(separateCommand[1]);
+                if (command.startsWith("mark")) {
+                    tasks.get(taskNumber - 1).markAsDone();
+                    System.out.println(" Marked as done: ");
+
+                } else if (command.startsWith("unmark")) {
+                    tasks.get(taskNumber - 1).markAsUndone();
+                    System.out.println(" Marked as not done yet: ");
+                }
+                writeLine(filePath);
+                System.out.println("   " + tasks.get(taskNumber - 1).toString());
+            } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("☹ OOPS!!! Invalid number");
+            }
+        } else if (command.startsWith("todo") || command.startsWith("deadline") || command.startsWith("event")
+                || command.startsWith("delete")) {
+            if (command.startsWith("todo")) {
+                try {
+                    String description = command.substring(5);
+                    if (description.length() == 0) {
+                        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    Task currTask = new ToDo(description);
+                    tasks.add(currTask);
+                    writeLine(currTask.toTxtString());
+                } catch (StringIndexOutOfBoundsException e) {
+                    throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                }
+            } else if (command.startsWith("deadline")) {
+                try {
+                    String[] parts = command.split("/by"); //   2/12/2019 1800
+                    String description = parts[0].substring(9).trim();
+                    if (description.length() == 0) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String byID = parts[1].trim();
```

Comment:
The naming of byID seems a bit confusing to the reader, would it be better to change it to dueDate?